### PR TITLE
Modular resource updates for python

### PR DESCRIPTION
**Pull request by mcvella**
Based on my experience building modular resources for python, here are changes I think should be made

### CODE REVIEW

File: docs/extend/modular-resources/_index.md [hunk 1]
```diff
@@ -440,6 +469,9 @@ To make this shell script executable, run the following command in your terminal
 sudo chmod +x <FILEPATH>/<FILENAME>
 ```
 
+You need to ensure any dependencies for your module (including the Viam SDK) are installed, as well.
+Your executable will be run by *viam-server* as root, so dependencies will need to be available to the root user.
```
Comment (npentrel) [inline docs/extend/modular-resources/_index.md:473]:
```suggestion
Your executable will be run by `viam-server` as root, so dependencies need to be available to the root user.
```

File: docs/extend/modular-resources/_index.md [hunk 2]
```diff
@@ -64,7 +64,9 @@ However, you can also create and expose new API types using modular resources.
 
 ## Use a modular resource with your robot
 
-Add a modular resource to your robot configuration in five steps:
+To add a modular resource to your robot configuration, you will sometimes use a modular resource that already exists (that someone else created).
+If this is the case, skip to [Save the executable](#make-sure-viam-server-can-access-your-executable)
+If you are creating your own modular resource, follow these 5 steps:
```
Comment (npentrel) [inline docs/extend/modular-resources/_index.md:69]:
```suggestion
If you are using an existing modular resource, you can skip to [Save the executable](#make-sure-viam-server-can-access-your-executable).
If you are creating your own modular resource, follow these steps:
```
- avoid future tense where possible
- shorten (for clarity)
- if it's important who created it we can add "that someone else created" back in
- where possible avoid specifying the number of steps. Easy to forget to update if the number changes.

**Review by npentrel — COMMENTED** [review]
Good changes - two suggestions for clarity.

File: docs/extend/modular-resources/_index.md [hunk 3]
```diff
@@ -399,6 +400,33 @@ from .my_base import MyBase
 Registry.register_resource_creator(Base.SUBTYPE, MyBase.MODEL, ResourceCreatorRegistration(MyBase.new_base, MyBase.validate_config))
 ```
 
+<file>main.py</file>
+``` python {class="line-numbers linkable-line-numbers"}
```
Comment (mcvella) [inline docs/extend/modular-resources/_index.md:404]:
```suggestion

``` python {class="line-numbers linkable-line-numbers"}
```

File: docs/extend/modular-resources/_index.md [hunk 4]
```diff
@@ -383,7 +384,7 @@ class MyBase(Base, Reconfigurable):
         return self.left.is_powered(extra=extra, timeout=timeout)[0] or self.right.is_powered(extra=extra, timeout=timeout)[0]
 ```
 
-<file>init.py</file>
+<file>__init.py__</file>
```
Comment (mcvella) [inline docs/extend/modular-resources/_index.md:387]:
```suggestion
<file>\__init.py__</file>
```

File: docs/extend/modular-resources/_index.md [hunk 5]
```diff
@@ -383,7 +384,7 @@ class MyBase(Base, Reconfigurable):
         return self.left.is_powered(extra=extra, timeout=timeout)[0] or self.right.is_powered(extra=extra, timeout=timeout)[0]
 ```
 
-<file>init.py</file>
+<file>\__init.py__</file>
```
Comment (mcvella) [inline docs/extend/modular-resources/_index.md:387]:
```suggestion
<file>`__init.py__`</file>
```

**Review by npentrel — APPROVED** [review]

**Pull request merged**